clarifai/rag.py

```diff
@@ -0,0 +1,122 @@
+from datetime import datetime
+
+import yaml
+from google.protobuf.struct_pb2 import Struct
+
+from clarifai.client.app import App
+from clarifai.client.model import Model
+from clarifai.client.user import User
+from clarifai.client.workflow import Workflow
+
+
+class RAG:
+  """
+    RAG is a class for Retrieval Augmented Generation.
+
+    Example:
+        >>> from clarifai.rag import RAG
+        >>> rag_agent = RAG()
+    """
+
+  def __init__(self,
+               workflow_url: str = None,
+               base_url: str = "https://api.clarifai.com",
+               pat: str = None,
+               **kwargs):
+    """
+        Initialize an empty or existing RAG.
+        """
+    w = Workflow(url=workflow_url, base_url=base_url, pat=pat)
+    self._prompt_workflow = w
+    self._app = App(app_id=w.app_id, base_url=w.base, pat=w.pat)
+    del w
+
+  def setup(self,
+            user_id: str = None,
+            llm_url: str = "https://clarifai.com/mistralai/completion/models/mistral-7B-Instruct",
+            base_workflow: str = "Text",
+            workflow_yaml_filename: str = 'prompter_wf.yaml',
```

Comment:
any reason on why we are restricting users to provide their yml(`prompt_template`) ?.

Reply:
Not sure if this comment could be made more clear to help answer this question:
> \## Generate a tmp yaml file for workflow creation

Maybe we could specify a `/tmp` location or something. I was concerned about permissions.

Reply:
Yes, but instead of `workflow_yaml_filename` maybe we could expose `prompt_template` arg ?

```
"prompt_template":
            "Context information is below:\n{data.hits}\nGiven the context information and not prior knowledge, answer the query.\nQuery: {data.text.raw}\nAnswer: "
    }
```

Reply:
Those are separate things. Notice that the workflow yaml file is _generated_, not user supplied.
I think at this stage we have no need to expose the `prompt_template`.